opensearch/src/main/java/org/opensearch/sql/opensearch/storage/scan/OpenSearchIndexScanBuilder.java

```diff
@@ -24,36 +24,39 @@
  * by delegated builder internally. This is to avoid conditional check of different push down logic
  * for non-aggregate and aggregate query everywhere.
  */
-public class OpenSearchIndexScanBuilder extends TableScanBuilder {
+public abstract class OpenSearchIndexScanBuilder extends TableScanBuilder {
 
   /**
    * Delegated index scan builder for non-aggregate or aggregate query.
    */
   @EqualsAndHashCode.Include
-  private TableScanBuilder delegate;
+  private PushDownQueryBuilder delegate;
 
   /** Is limit operator pushed down. */
   private boolean isLimitPushedDown = false;
 
-  @VisibleForTesting
-  OpenSearchIndexScanBuilder(TableScanBuilder delegate) {
-    this.delegate = delegate;
+  /**
+   * Constructor used during query execution.
+   */
+  protected OpenSearchIndexScanBuilder(OpenSearchRequestBuilder requestBuilder) {
+    this.delegate = new OpenSearchIndexScanQueryBuilder(requestBuilder);
+
   }
 
   /**
-   * Initialize with given index scan.
-   *
-   * @param indexScan index scan to optimize
+   * Constructor used for unit tests.
    */
-  public OpenSearchIndexScanBuilder(OpenSearchIndexScan indexScan) {
-    this.delegate = new OpenSearchIndexScanQueryBuilder(indexScan);
+  protected OpenSearchIndexScanBuilder(PushDownQueryBuilder translator) {
+    this.delegate = translator;
   }
 
   @Override
   public TableScanOperator build() {
-    return delegate.build();
+    return createScan(delegate.build());
```

Comment:
what is the difference of build and createScan?

Reply:
`OpenSearchIndexScanBuilder.createScan` -- create an OpenSearchIndexScan based on the provided  OpenSearchRequest. 

`OpenSearchIndexScanBuilder.build` -- build an OpenSearchRequest and an OpenSearchIndexScan based on it.

They will be merged as part of making `OpenSearchIndexScanBuilder` inner class of `OpenSearchIndexScan`. 

Related to the [other](https://github.com/opensearch-project/sql/pull/1666#discussion_r1210810444) discussion about builders.